packages/ketcher-react/src/script/editor/tool/atom.ts

```diff
@@ -65,174 +65,199 @@ class AtomTool {
-    this.editor.hover(null)
-    this.editor.selection(null)
-    const ci = this.editor.findItem(event, ['atoms', 'functionalGroups'])
-    if (
-      ci &&
-      ci.map === 'functionalGroups' &&
-      functionalGroups &&
-      FunctionalGroup.isContractedFunctionalGroup(ci.id, functionalGroups)
-    ) {
-      const action = new Action()
-      const selectedSgroup = sgroups.get(ci.id)
-      const sGroupAtoms = SGroup.getAtoms(molecule, selectedSgroup?.item)
-      const [firstAtom, ...atoms] = sGroupAtoms
-      const atomNeighbours = molecule.atomGetNeighbors(firstAtom)
-      const extraNeighbour = atomNeighbours?.some(
-        (atom) => !sGroupAtoms.includes(atom.aid)
-      )
-      if (extraNeighbour) {
-        action.mergeWith(fromSgroupDeletion(struct, ci.id))
-        action.mergeWith(fromFragmentDeletion(struct, { atoms: atoms }))
-        action.mergeWith(
-          fromAtomsAttrs(struct, firstAtom, this.atomProps, true)
-        )
-      } else {
-        const firstAtomPp = struct.atoms.get(firstAtom)?.a.pp
-        action.mergeWith(
-          fromFragmentDeletion(struct, {
-            atoms: SGroup.getAtoms(molecule, selectedSgroup?.item),
-            bonds: SGroup.getBonds(molecule, selectedSgroup?.item)
-          })
-        )
-        action.mergeWith(fromAtomAddition(struct, firstAtomPp, this.atomProps))
+    const {
+      editor,
+      editor: {
+        render: {
+          ctab: {
+            molecule,
+            molecule: { functionalGroups }
+          }
+        }
       }
-      this.editor.update(action)
-    }
-    const atomResult: Array<number> = []
-    const result: Array<number> = []
-    if (ci && functionalGroups.size && ci.map === 'atoms') {
-      const atomId = FunctionalGroup.atomsInFunctionalGroup(
+    } = this
+
+    editor.hover(null)
+    editor.selection(null)
+
+    const eventMaps = ['atoms', 'functionalGroups']
+    const ci = editor.findItem(event, eventMaps)
+
+    if (ci?.map === 'atoms') {
+      const atomId = ci.id
+
+      const fgId = FunctionalGroup.findFunctionalGroupByAtom(
         functionalGroups,
-        ci.id
+        atomId
       )
-      if (atomId !== null) atomResult.push(atomId)
-    }
-    if (atomResult.length > 0) {
-      for (const id of atomResult) {
-        const fgId = FunctionalGroup.findFunctionalGroupByAtom(
-          functionalGroups,
-          id
-        )
-        if (fgId !== null && !result.includes(fgId)) {
-          result.push(fgId)
-        }
+
+      if (fgId !== null) {
+        editor.event.removeFG.dispatch({ fgIds: [fgId] })
+        return
       }
-      this.editor.event.removeFG.dispatch({ fgIds: result })
-      return
     }
-    if (!ci) {
-      // ci.type == 'Canvas'
-      this.dragCtx = {}
-    } else if (ci.map === 'atoms') {
-      this.dragCtx = { item: ci }
+
+    const ciFunctionalGroupName =
+      ci?.map === 'functionalGroups'
+        ? molecule.functionalGroups.get(ci?.id)?.name
```

Comment:
could we write here "ci?.map === 'functionalGroups' && molecule.functionalGroups.get(ci?.id)?.name"?